sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java

```diff
@@ -348,11 +342,11 @@ private void checkPushInActivityIntent(final Activity activity, final Intent int
             AppCenterLog.error(LOG_TAG, "Push.checkLaunchedFromNotification: intent may not be null");
             return;
         }
+        mActivity = activity;
         postOnUiThread(new Runnable() {
 
             @Override
             public void run() {
-                mActivity = activity;
                 checkPushInIntent(intent);
```

Comment:
The post was making sure activity is not null or mismatching the activity where we received the push in foreground.
So this solution is simplifying but to avoid regression you need to pass (final) activity (not mActivity) to checkPushInIntent, the activity snapshot basically for that callback.

Reply:
This is a literal description of what to do 😄